src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs

```diff
@@ -323,6 +323,28 @@ public static Vector2 Create(ReadOnlySpan<float> values)
         [Intrinsic]
         internal static Vector2 CreateScalarUnsafe(float x) => Vector128.CreateScalarUnsafe(x).AsVector2();
 
+        /// <summary>
+        /// Returns the z-value of the cross product of two vectors.
+        /// Since the Vector2 is in the x-y plane, a 3D cross product only produces the z-value.
+        /// </summary>
+        /// <param name="value1">The first vector.</param>
+        /// <param name="value2">The second vector.</param>
+        /// <returns>The value of the z-coordinate from the cross product.</returns>
+        /// <remarks>
+        /// Return z-value = value1.X * value2.Y - value1.Y * value2.X
+        /// <see cref="Cross"/> is the same as taking the <see cref="Dot"/> with the second vector
+        /// that has been rotated 90-degrees.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float Cross(Vector2 value1, Vector2 value2)
+        {
+            //return value1.X * value2.Y - value1.Y * value2.X;
+
+            Vector128<float> mul = value1.AsVector128Unsafe() *
+                Vector128.Shuffle(value2.AsVector128Unsafe(), Vector128.Create(1, 0, 1, 0));
+            return (mul - Vector128.Shuffle(mul, Vector128.Create(1, 0, 1, 0))).ToScalar();
```

Comment:
Similar here. It is preferable to use `AsVector128` or to guarantee the zeroing of the upper elements by using `-1` as the index in the shuffle. It helps avoid pessimization and should be naturally part of the underlying codegen without inserting additional instructions